src/Paket.Core/NugetConvert.fs

```diff
@@ -228,21 +229,35 @@ let createPackageRequirement (packageName, version, restrictions) sources depend
 let createDependenciesFileR (rootDirectory : DirectoryInfo) nugetEnv mode =
     
     let dependenciesFileName = Path.Combine(rootDirectory.FullName, Constants.DependenciesFileName)
-    
-    let allVersions =
+
+    let allVersionsById =
 
         nugetEnv.NugetProjectFiles
         |> Seq.collect (fun (_,c) -> c.Packages)
         |> Seq.groupBy (fun p -> p.Id)
+
+    let allVersionsByName =
```

Comment:
the `allVersionsByName` and `allVersionsById` are bit misleading, actually id = name for the packages. maybe we could just extract the set of all versions and set of all targetFramwork seperately?